### PR TITLE
Capture log files during selfhost setup and cleanup

### LINKED ISSUE #1112: Starting self-host service should capture logs for the .exe's it runs
Currently the script to start the self-hosted WCF service uses 'start' to execute the certificate generator .exe elevated.  Errors occurring during cert generation are ignored and will appear only in this window.  I recommend the elevated CMD should capture the output of each .exe it invokes into a single log file.  And the CMD that is waiting for the service to start should echo that logfile to the console.

This way, we will be able to see what happened on CI machines running self-host.  I believe certificate generation may be failing on a self-host CI, but there is no capture of what went wrong.  The fact [ConditionaFact] thinks no client certificate is installed might be due to certificate generation failure, and having this log in the console for the CI run would tell us.

### PR DESCRIPTION
**Pull request by roncain**
To facilitate understanding failures when selfhost outerloops
run in the lab, capture the output of each of the separate
scripts and exe's and log them to msbuild output.

Fixes #1112

### CODE REVIEW

**Comment by roncain** [comment]
@dotnet-bot test outerloop selfhost Windows_NT

**Comment by roncain** [comment]
Reviewers: trying this in CI shows that the information logged to MsBuild by this PR is not making it to the console output.  I need to investigate a bit more to see how to make that happen.  Review feedback still welcome.

**Comment by roncain** [comment]
@dotnet-bot test outerloop selfhost Windows_NT

**Comment by roncain** [comment]
Reviewers: to see what this PR does, look at the details for the CI selfhost OuterLoop run.  Open the console output and view the entire console output.  Search for the string "------" to see our self-host autostart and autoclose.

I've also tried this using a non-elevated window, and the self-host service is now properly closed.

File: src/System.Private.ServiceModel/tools/scripts/CleanUpWCFSelfHostedSvc.cmd [hunk 1]
```diff
@@ -1,44 +1,67 @@
...
+
 REM
 echo Cleanup test services
 REM  Note that we always return successful code in clean up routine because clean up errors are not fatal
 
-TASKKILL /F /IM SelfHostedWCFService.exe
-REM error 128 will be returned if The process not found
-if NOT [%ERRORLEVEL%]==[0]  if NOT [%ERRORLEVEL%]==[128] (
-	echo Warning: An error occurred while killing the SelfHostedWCFService.exe.
-	)
```
Comment (hongdai) [inline src/System.Private.ServiceModel/tools/scripts/CleanUpWCFSelfHostedSvc.cmd]: I see.  The taskkill needed to be elevated as well.

**Comment by hongdai** [comment]
LGTM. Great logging details to help identify issues. Thanks Ron!

**Comment by roncain** [comment]
@dotnet-bot test outerloop Windows_NT

**Pull request merged**